index.js

```diff
@@ -135,22 +135,32 @@ if (platform === 'linux') {
 // TODO: Use https://github.com/sindresorhus/is-unicode-supported when targeting Node.js 10.
 const figures = platform === 'win32' ? fallback : main;
 
-const fn = string => {
-	if (figures === main) {
-		return string;
+const isFallbackFigure = ([key, mainValue]) => figures[key] !== mainValue;
+const getFigureRegExp = ([key, mainValue]) => [figures[key], new RegExp(escapeStringRegexp(mainValue), 'g')];
+
+let replacements = [];
+const getReplacements = () => {
+	if (replacements.length !== 0) {
+		return replacements;
 	}
 
-	for (const [key, value] of Object.entries(main)) {
-		if (value === figures[key]) {
-			continue;
-		}
+	replacements = Object.entries(main)
+		.filter(isFallbackFigure)
+		.map(getFigureRegExp);
+	return replacements;
+};
 
-		string = string.replace(new RegExp(escapeStringRegexp(value), 'g'), figures[key]);
+const replaceCharToFallback = (string, [fallbackValue, mainRegExp]) => string.replace(mainRegExp, fallbackValue);
+
+// On Windows, substitute non-fallback to fallback figures
+const replaceCharsToFallback = string => {
+	if (figures === main) {
+		return string;
 	}
 
-	return string;
+	return getReplacements().reduce(replaceCharToFallback, string);
```

Comment:
I prefer not using `.reduce` in general.

Reply:
Fixed in eb5a30495baabdeb4bb5340625abae9bd3923099